### ISSUE TITLE
Adapted create-react-app does not work

### ISSUE THREAD

**Issue by freddi301** [liferay/liferay-frontend-projects#600]
### Issue type

-   [ ] :thinking: Question
-   [x] :bug: Bug report
-   [ ] :gift: Feature request
-   [ ] :woman_shrugging: Other

### Description

**Current behavior:** `TypeError: Cannot read property 'adaptStaticURL' of undefined` when portlet is loaded in page

**Repro instructions (if applicable):**
- create-react-app
- yo liferay-js:adapt
- deploy

**Other information (environment, versions etc):**
- lifera-portal: 7.7.3-ga8 CE
- node: 12.18.4
- windows 10
- yo liferay-js:adapt 4.2.0
- react-scripts 4.0.3

**Comment by freddi301** [comment]
`_ADAPT_RT_` is undefined

**Comment by bryceosterhaus** [comment]
@freddi301 Could you provide some more information? Node version, Liferay DXP version, and generator-liferay-js version would all be helpful.

I just tested locally and I am not seeing any issues.

![Screen Shot 2021-06-28 at 11 44 13 AM](https://user-images.githubusercontent.com/6843530/123687777-2b223e80-d806-11eb-8b8b-fb1ab481bb75.png)

**Comment by freddi301** [comment]
Hi, I added versions to the description.
I tried on CE but target is DXP.
Which version did you use?
Thank you

**Comment by freddi301** [comment]
removing from project
```javascript
import spritemap from "@clayui/css/lib/images/icons/icons.svg"
```
it works

**Comment by bryceosterhaus** [comment]
Oh nice! Also, if you need to access the spritemap within DXP you can try using something like 

```js
const spritemap = `${Liferay.ThemeDisplay.getPathThemeImages()}/clay/icons.svg`;
```

Going to close this issue but feel free to respond or re-open if you have any other issues

**Comment by freddi301** [comment]
Thank you, maybe this issue can be useful as reference for when static resource imports do not work with adapt

**Comment by bryceosterhaus** [comment]
cc: @izaera, do you know if we have investigated static resource imports? Would it be possible with our loader to do something where the imported static resource gives a path to the resource on the server?

**Comment by izaera** [comment]
>  do you know if we have investigated static resource imports? 

The fact is that adaptation is provided:

1) As a way to transition from native framework projects to Liferay ones (those generated with the generator, not with `create-react-app` or `@liferay/cli`)
2) As a way to reuse an existing codebase
3) As a way to write platform agnostic projects than can be deployed to several platforms

This is because adaptation is a fragile process partially based on heuristics, since we are transforming a JS bundle output by webpack to be deployable in Liferay. Because a webpack bundle is, by definition, and output artifact, it's impossible to do all the things you would be able to do with proper source (input) artifacts, like the source code in true Liferay projects.

Regarding static resources, we transform some URLs inside CSS files, for example, but it's a best effort feature of the tool, not anything fully supported or guaranteed. There may even be corner cases where it may fail (when you configure CDNs or proxies) as that information is not available during build time.

So, in a nutshell, the adaptation does what it can and the rest is the developer's responsibility.

In this case, however, the cause may be different (that "`_ADAPT_RT_` is undefined" is suspicious), so I'll reopen this to investigate it :+1:

**Comment by kresimir-coko** [comment]
> In this case, however, the cause may be different (that "_ADAPT_RT_ is undefined" is suspicious), so I'll reopen this to investigate it 👍

@izaera are you going to have time to investigate this soon, or should we close it?

**Comment by izaera** [comment]
I'll close it, since it hasn't happened again and I doubt we will have time to investigate.
